docs/src/test-components-js.md

```diff
@@ -195,93 +195,168 @@ npm run test-ct
+</TabItem>
+<TabItem value="vue">
 
-Let's say you'd like to test following component:
+```js
+test('props', async ({ mount }) => {
+  const component = await mount(Component, { props: { msg: 'greetings' } });
+});
+```
 
-```js title="input-media.tsx"
-import React from 'react';
+</TabItem>
 
-export const InputMedia: React.FC<{
-  // Media is a complex browser object we can't send to Node while testing.
-  onChange: (media: Media) => void,
-}> = ({ onChange }) => {
-  return <></> as any;
-};
+</Tabs>
+
+### callbacks / events
+
+Provide callbacks/events to a component when mounted.
+
+<Tabs
+  defaultValue="react"
+  values={[
+    {label: 'React', value: 'react'},
+    {label: 'Solid', value: 'solid'},
+    {label: 'Svelte', value: 'svelte'},
+    {label: 'Vue', value: 'vue'},
+  ]
+}>
+
+<TabItem value="react">
+
+```js
+test('callback', async ({ mount }) => {
+  const component = await mount(<Component callback={() => {}} />);
+});
 ```
 
-Create a story file for your component:
+</TabItem>
+<TabItem value="solid">
 
-```js title="input-media.story.tsx"
-import React from 'react';
-import InputMedia from './import-media';
+```js
+test('callback', async ({ mount }) => {
+  const component = await mount(<Component callback={() => {}} />);
+});
+```
 
-export const InputMediaForTest: React.FC<{
-  onMediaChange: (mediaName: string) => void,
-}> = ({ onMediaChange }) => {
-  // Instead of sending a complex `media` object to the test, send the media name.
-  return <InputMedia onChange={media => onMediaChange(media.name)} />;
-};
-// Export more stories here.
+</TabItem>
+<TabItem value="svelte">
+
+```js
+test('event', async ({ mount }) => {
+  const component = await mount(Component, { on: { callback() {} } });
+});
 ```
 
-Then test the component via testing the story:
+</TabItem>
+<TabItem value="vue">
 
-```js title="input-media.test.spec.tsx"
-test('changes the image', async ({ mount }) => {
-  let mediaSelected: string | null = null;
+```js
+test('event', async ({ mount }) => {
+  const component = await mount(Component, { on: { callback() {} } });
+});
+```
 
-  const component = await mount(
-    <InputMediaForTest
-      onMediaChange={mediaName => {
-        mediaSelected = mediaName;
-        console.log({ mediaName });
-      }}
-    />
-  );
-  await component
-    .getByTestId('imageInput')
-    .setInputFiles('src/assets/logo.png');
+</TabItem>
 
-  await expect(component.getByAltText(/selected image/i)).toBeVisible();
-  await expect.poll(() => mediaSelected).toBe('logo.png');
+</Tabs>
+
+### children / slots
+
+Provide children/slots to a component when mounted.
```

Comment:
Perhaps an example with a child component (as opposite to text child) would be nice?

Reply:
Component as slot/child is not yet available across all frameworks: https://github.com/microsoft/playwright/issues/15900. Perhaps it's better to update this later on?

Reply:
Yeah, makes sense.